packages/fxa-settings/src/components/ButtonDownloadRecoveryKeyPDF/index.tsx

```diff
@@ -0,0 +1,147 @@
+  findOutMoreHeading: string;
+}
+
+interface ButtonDownloadRecoveryKeyPDFProps {
+  navigateForward?: () => void;
+  recoveryKeyValue: string;
+  viewName: string;
+}
+
+export const ButtonDownloadRecoveryKeyPDF = ({
+  navigateForward,
+  recoveryKeyValue,
+  viewName,
+}: ButtonDownloadRecoveryKeyPDFProps) => {
+  const account = useAccount();
+  const email = account.primaryEmail.email;
+  const keyCreated = Date.now();
+  const currentLanguage = navigator.language;
+  const ftlMsgResolver = useFtlMsgResolver();
+
+  const keyDateFallback = Intl.DateTimeFormat('default', {
+    dateStyle: 'medium',
+  }).format(new Date(keyCreated));
+
+  const keyDateFluent = getLocalizedDate(
+    keyCreated,
+    LocalizedDateOptions.MediumDateStyle
+  );
+
+  const localizedText: LocalizedRecoveryKeyPdfContent = {
+    heading: ftlMsgResolver.getMsg(
+      'recovery-key-pdf-heading',
+      'Account Recovery Key'
+    ),
+    dateGenerated: ftlMsgResolver.getMsg(
+      'recovery-key-pdf-download-date',
+      `Generated: ${keyDateFallback}`,
+      { date: keyDateFluent }
+    ),
+    keyLegend: ftlMsgResolver.getMsg(
+      'recovery-key-pdf-key-legend',
+      'Account Recovery Key'
+    ),
+    instructions: ftlMsgResolver.getMsg(
+      'recovery-key-pdf-instructions',
+      'This key allows you to recover your encrypted browser data (including passwords, bookmarks, and history) if you forget your password. Store it in a place you’ll remember. '
+    ),
+    storageHeading: ftlMsgResolver.getMsg(
+      'recovery-key-pdf-storage-ideas-heading',
+      'Places to store your key'
+    ),
+    storageIdeaFolder: ftlMsgResolver.getMsg(
+      'flow-recovery-key-download-storage-ideas-folder-v2',
+      'Folder on secure device'
+    ),
+    storageIdeaCloud: ftlMsgResolver.getMsg(
+      'flow-recovery-key-download-storage-ideas-cloud',
+      'Trusted cloud storage'
+    ),
+    storageIdeaPrint: ftlMsgResolver.getMsg(
+      'flow-recovery-key-download-storage-ideas-print-v2',
+      'Printed physical copy'
+    ),
+    storageIdeaPwdManager: ftlMsgResolver.getMsg(
+      'flow-recovery-key-download-storage-ideas-pwd-manager',
+      'Password manager'
+    ),
+    findOutMoreHeading: ftlMsgResolver.getMsg(
+      'recovery-key-pdf-support',
+      'Learn more about your account recovery key'
+    ),
+  };
+
+  const getFilename = () => {
+    const date = new Date().toISOString().split('T')[0];
+    // Windows has a max directory length of 260 characters (including path)
+    // filename should be kept much shorter (maxLength is arbitrary).
+    const maxLength = 75;
+    const prefix = 'Firefox-Recovery-Key';
+    let email = account.primaryEmail.email;
+    let filename = `${prefix}_${date}_${email}.pdf`;
+
+    if (filename.length > maxLength) {
+      const lengthWithoutEmail = filename.length - email.length;
+      email = email.slice(0, maxLength - lengthWithoutEmail);
+      filename = `${prefix}_${date}_${email}.pdf`;
+    }
```

Comment:
If I understand what's happening here, then my suggestion is to use `.substring` to limit the length and then append '.pdf', i.e. `` `${prefix}_${date}_${email}`.substring(0, 71) ``

Reply:
@pdehaan That's a good question, and I haven't found a clear answer. Some characters (`+`, `@`) could potentially be problematic. 

TBH, I'd rather omit the email from the filename entirely - it would avoid all the trouble with truncating and validation. I'll check in with UX about simplifying to `${prefix}_${date}.pdf`